katapult.go

```diff
@@ -17,26 +19,26 @@ const (
 
 var DefaultURL = &url.URL{Scheme: "https", Host: "api.katapult.io"}
 
-type Opt func(c *Client) error
+type Opt func(c *Client, h *http.Client) error
```

Comment:
I think `h` variable name could be expanded a bit to indicate it is given the default `http.Client` only, and not the custom one if the `WithHTTPClient()` option is used.

It's kinda horrible, but something like this maybe?

```suggestion
type Opt func(c *Client, defaultHTTPClient *http.Client) error
```